src/track/trackrecord.cpp

```diff
@@ -108,6 +108,29 @@ bool copyIfNotEmpty(
 
 } // anonymous namespace
 
+bool TrackRecord::replaceMetadataFromSource(
+        TrackMetadata&& importedMetadata,
+        const QDateTime& metadataSynchronized) {
+    bool modified = false;
+    if (getMetadata() != importedMetadata) {
+        refMetadata() = std::move(importedMetadata);
```

Comment:
This std::move is redundant. We have a rvalue setter. Let's use it. 
```suggestion
        setMetadata(importedMetadata);
```

Reply:
This code is hard to read. Let's use setMetadata()

Reply:
Good catch. setMetadata() also uses a universal reference and std::forward.

Reply:
But we need to keep std::move() to ensure that we don't invoke an overload that accepts `const TrackMetadata&`. Currently there is none, but otherwise whenever it is added in the future this code would behave differently without std::move(). It is needed to express out intend.

Reply:
See also: https://stackoverflow.com/a/32691106

Reply:
Ah OK, I did not know that.

Reply:
C++ has become too complex over time. Ideally the compiler would detect those pitfalls. I can't remember them all, too.